Optimize coco for performance

cyanguwa opened the pull request:
This version of cocoapi offers optimizations in the C++ extensions and about 3x speedup. It also provides multi-process support and when run in parallel, 8x speedup in our tests. The multiple processes parallelize across the category dimension, and it is the user's responsibility to spawn up processes and reduce results afterwards. Users can choose to run the evaluation jobs synchronously or asynchronously to their training jobs.

cyanguwa committed:
optimize coco for performance and add multi-process support

cliffwoolley commented:
I understand this (among other things) incorporates the purpose of https://github.com/NVIDIA/cocoapi/pull/12 .  I will close that one in favor of this one.

cliffwoolley commented:
@cyanguwa - It's kind of a nit, but would you mind please following the formatting of the original files?  At least for tabs vs spaces.  It would make the diff a lot easier to read.

cyanguwa committed:
change tabs to 2-space indents; remove using namespace std/simdjson

cyanguwa commented:
Hi @cliffwoolley, please take a look at the new commit (cad33d1) now. I changed the tabs back to two spaces and also removed the use of 'using namespace std/simdjson' along the way. 

Thanks!
